modules/pipeline/services/pipelinesvc/create_task.go

```diff
@@ -188,11 +188,27 @@ func (s *PipelineSvc) judgeTaskExecutor(action *pipelineyml.Action, actionSpec *
 func calculateNormalTaskResources(action *pipelineyml.Action, actionDefine *diceyml.Job) apistructs.PipelineAppliedResources {
 	defaultRes := apistructs.PipelineAppliedResource{CPU: conf.TaskDefaultCPU(), MemoryMB: conf.TaskDefaultMEM()}
 	return apistructs.PipelineAppliedResources{
-		Limits:   calculateNormalTaskLimitResource(action, actionDefine, defaultRes),
+		Limits:   calculateOversoldTaskLimitResource(calculateNormalTaskLimitResource(action, actionDefine, defaultRes), defaultRes),
 		Requests: calculateNormalTaskRequestResource(action, actionDefine, defaultRes),
 	}
 }
 
+func calculateOversoldTaskLimitResource(limits apistructs.PipelineAppliedResource, defaultRes apistructs.PipelineAppliedResource) apistructs.PipelineAppliedResource {
+	maxCPU := limits.CPU
+	maxMemoryMB := limits.MemoryMB
+	// Cpu is usually be wasted, if action and action define cpu is lower than default, use default cpu
+	if maxCPU < defaultRes.CPU {
```

Comment:
where is defaultRes come from

Reply:
Maybe you should multiply the default oversold rate.
If result larger than max_allowed_oversold_cpu, than use max_allowed_oversold_cpu.